File: android/lib/core/clipboard_model.dart
/// Clipboard Event Message Model (Phase 4)
class ClipboardEventMessage {
  final String eventId;
  final String originDeviceId;
  final int timestamp;
  final String contentType;
  final String content;

  ClipboardEventMessage({
    required this.eventId,
    required this.originDeviceId,
    required this.timestamp,
    this.contentType = 'text/plain',
    required this.content,
  });

  Map<String, dynamic> toJson() => {
    'eventId': eventId,
    'originDeviceId': originDeviceId,
    'timestamp': timestamp,
    'contentType': contentType,
    'content': content,
  };

  factory ClipboardEventMessage.fromJson(Map<String, dynamic> json) {
    return ClipboardEventMessage(
      eventId: json['eventId'] as String? ?? '',
      originDeviceId: json['originDeviceId'] as String? ?? '',
      timestamp: (json['timestamp'] as num?)?.toInt() ?? 0,
      contentType: json['contentType'] as String? ?? 'text/plain',
      content: json['content'] as String? ?? '',
    );
  }
}

File: windows/Hinge.Core/ClipboardModel.cs
using System.Text.Json.Serialization;

namespace Hinge.Core;

public class ClipboardEventMessage
{
    [JsonPropertyName("eventId")]
    public string EventId { get; set; } = Guid.NewGuid().ToString();

    [JsonPropertyName("originDeviceId")]
    public string OriginDeviceId { get; set; } = string.Empty;

    [JsonPropertyName("timestamp")]
    public long Timestamp { get; set; } = DateTimeOffset.UtcNow.ToUnixTimeMilliseconds();

    [JsonPropertyName("contentType")]
    public string ContentType { get; set; } = "text/plain";

    [JsonPropertyName("content")]
    public string Content { get; set; } = string.Empty;
}
